CIP-XXXX/README.md

```diff
@@ -0,0 +1,1597 @@
+
+### Appendix
+
+This appendix contains additional schemas for types that are used in the APIs.
+
+#### Connection API Data Types
+
+##### Extensions
+
+```
+{
+  "type": "object",
+  "title": "Extensions",
+  "properties": {
+    "extensions": {
+      "type": "array"
+      "items": {
+          {
+            "type": "object",
+            "title": "Extension",
+            "properties": {
+              "cip": {
+                "type": "number"
+              },
+              "version: {
+                "$ref": "#/appendix/SemVer"
+              }
+            },
+            "required": ["cip"],
+          }
+      }
+    }
+  },
+  "required": ["extensions"],
+}
+```
+
+An extension is an object with a "cip" field that describes a CIP number extending the API, and an optional version specified according to SemVer. For example:
+
+```
+{ "cip": 30 }
+```
+
+
+##### DataSignature
+
+```
+{
+  "type": "object",
+  "title": "DataSignature",
+  "properties": {
+    "signature": {
+      "title": "Ed25519Signature",
+      "type": "string",
+      "format": "hex",
+      "pattern": "^([0-9a-f][0-9a-f]){64}$"
+    },
+    "key": {
+      "title": "Ed25519PublicKey",
+      "type": "string",
+      "format": "hex",
+      "pattern": "^([0-9a-f][0-9a-f]){32}$"
+    }
+  },
+  "required": ["signature", "key"],
+}
+```
+
+An object representing some data that has been signed. It contains 2 fields: `signature` which contains the signed data, and `key` which contains the derived Ed25519 PubKey used to sign the data.
+
+##### SemVer
+
+```
+{
+  "type": "object",
+  "title": "SemVer",
+  "properties": {
+    "major": {
+      "title": "Major",
+      "type": "number",
+    },
+    "minor": {
+      "title": "Minor",
+      "type": "number",
+    },
+    "patch": {
+      "title": "Patch",
+      "type": "number",
+    }
+  },
+  "required": ["major", "minor", "patch"],
+}
+```
+
+#### Errors
+
+This section lists the possible errors the wallet connector may return. Each error is specified by an error code, and an optional info string describing the cause of the error. Below each error we list names for all the error codes, and a brief description of when they should be used. Other than when the spec dictates the use of a specific error code, wallets can choose the error code they deem more applicable to the situation. Wallets can also extend these errors with additional codes should they feel the need to.
+
+##### APIError
+
+```
+{
+  "type": "object",
+  "title": "APIError",
+  "properties": {
+    "code": {
+      "type": "number",
+      "title": "APIErrorCode"
+      "enum": [-1, -2, -3, -4, -5]
+    },
+    "info": {
+      "type": "string"
+    }
+  },
+  "required": ["code"],
+}
+```
+
+*[Why are these negative? Seems arbitrary, is it too breaking to change them?]*
```

Comment:
> *[Why are these negative? Seems arbitrary, is it too breaking to change them?]*

There are some conventions that use negative numbers for error codes I believe. Probably best to keep the same to avoid any unnecessary changes from CIP-30.